### PR TITLE
feat: added product-listing page

### PR DESCRIPTION
**Pull request by partha8**
## files to check
- [src/pages/Product-Listing/ProductListing.js](https://github.com/partha8/algern-clothing-react/compare/product-listing?expand=1#diff-7707fc2a50c8ed3ec5f3b6381858aff9436d7dc3924b4238e2513982882d4a8c)
- [src/reducers/productsReducer.js](https://github.com/partha8/algern-clothing-react/compare/product-listing?expand=1#diff-1e72764740711c06e4f3cf410674a7d5a3bb67e5236ee138c5019d857f828e3a)

## Note
The filter component is commented out on the product-listing page. I will add it in future implementations.

### CODE REVIEW

File: src/pages/Product-Listing/ProductListing.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              image,
+              addedToWishlist,
+              addedToCart,
+            } = product;
+            return (
+              <div key={id} className="card card-vertical">
+                <div className="image-container-vert">
+                  <img
+                    className="img responsive-image product-image"
+                    src={image.src}
```
Comment (vrinda-mahajan) [inline src/pages/Product-Listing/ProductListing.js]: Suggestion: You can move your product card into a separate component.

File: src/pages/Product-Listing/ProductListing.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          {productsList.map((product) => {
+            const {
+              id,
+              name,
+              price,
+              marker,
+              image,
+              addedToWishlist,
+              addedToCart,
+            } = product;
```
Comment (Maianki) [inline src/pages/Product-Listing/ProductListing.js]: You can also destructure image here For e.g ```const {
              id,
              name,
              price,
              marker,
              image:{src:imageSrc,alt:imageAlt}, 
              addedToWishlist,
              addedToCart,
            } = product;```  or by simply doing ```const {
              id,
              name,
              price,
              marker,
              image:{src,alt}, 
              addedToWishlist,
              addedToCart,
            } = product;```

**Review by Maianki — APPROVED** [review]
Your code looks good to merge. Has added a small suggestion.
Things that I liked about your code:
- Separate folder for context and reducers.

File: src/context/StateProvider.js [hunk 3]
```diff
@@ -9,24 +17,33 @@ const StateProvider = ({ children }) => {
...
   );
 
+  useEffect(() => {
+    axios.get("/api/products").then((response) => {
+      productsDispatch({
+        type: "SET_PRODUCTS",
+        payload: response.data.products,
+      });
+    });
+  }, []);
```
Comment (vrinda-mahajan) [inline src/context/StateProvider.js]: Suggestion:- It is good to use async-await for API calls.

**Review by vrinda-mahajan — APPROVED** [review]
Added a few suggestions. The rest looks good.

File: src/pages/Product-Listing/ProductListing.js (hunk 2 above)
Comment (partha8) [reply src/pages/Product-Listing/ProductListing.js]: Yup, i will do that. Thank you.

File: src/pages/Product-Listing/ProductListing.js (hunk 1 above)
Comment (partha8) [reply src/pages/Product-Listing/ProductListing.js]: I will do that. Thank you @vrinda-mahajan for the review

File: src/context/StateProvider.js (hunk 3 above)
Comment (partha8) [reply src/context/StateProvider.js]: Yup, already on it. Thank you!

File: src/context/StateProvider.js [hunk 4]
```diff
@@ -9,24 +17,33 @@ const StateProvider = ({ children }) => {
...
 
-  const [{ cart, wishlist, productList }, productDispatch] = useReducer(
-    productReducer,
+  const [{ cart, wishlist, productsList }, productsDispatch] = useReducer(
+    productsReducer,
     initialProductState
   );
 
+  useEffect(() => {
+    axios.get("/api/products").then((response) => {
```
Comment (vrx29) [inline src/context/StateProvider.js]: do api calls in try catch it will help in debugging if you get any errors

Comment (partha8) [reply src/context/StateProvider.js]: I will do it. Thank you for the review!

**Pull request merged**